### ISSUE TITLE
Problematic spacing in citation

### ISSUE THREAD

**Issue by nickw2066** [zepinglee/citeproc-lua#44]
**Describe the bug**
There are unnecessary newlines in the citation.

**Additional information**
- TeX distribution: `MacTeX 2023`, `LuaHBTeX, Version 1.17.0 (TeX Live 2023)`
- Package `citation-style-language` version: v0.4.4
- LaTeX engine: `lualatex`

**To Reproduce**
```tex
% !TEX encoding = UTF-8 Unicode
\documentclass[12pt,oneside]{article}

\usepackage{citation-style-language}
\cslsetup{style = turabian-fullnote-bibliography}
\addbibresource{myBib.json}

\begin{document}

Asumsi\cite{Hiebert2006-Hermen}

\end{document}
```

JSON:
```
[
    {
        "id": "Hiebert2006-Hermen",
        "type": "chapter",
        "author": [
            {
                "family": "Hiebert",
                "given": "Robert J. V."
            }
        ],
        "editor": [
            {
                "family": "Kraus",
                "given": "Wolfgang"
            }
        ],
        "collection-editor": [
            {
                "family": "Peters",
                "given": "Melvin K. H."
            }
        ],
        "title": "The Hermeneutics of Translation in the Septuagint of Genesis",
        "source": "EBSCO",
        "collection-title": "Society of Biblical Literature Septuagint and Cognate Studies",
        "container-title": "Septuagint Research: Issues and Challenges in the Study of the Greek Jewish Scriptures",
        "page": "85-103",
        "publisher-place": "Atlanta",
        "publisher": "Society of Biblical Literature",
        "issued": {
            "date-parts": [
                [
                    "2006"
                ]
            ]
        }
    }
]

```

**Screenshots**
Output:
[mwe.pdf](https://github.com/zepinglee/citeproc-lua/files/12789429/mwe.pdf)

**Comment by hvoss49** [comment]
With the style from https://www.zotero.org/styles/turabian-fullnote-bibliography and the current github version it works well with any engine.

**Comment by zepinglee** [comment]
This bug is actually the same as <https://github.com/zepinglee/citeproc-lua/issues/42> and it's fixed in 73d3ad3.

In the provided JSON data, there are several non-breaking spaces in `container-title` and `collection-title` (see below). They are incorrectly converted to new lines in citeproc-lua v0.4.4.

<img width="916" alt="Screenshot 2023-10-04 at 15 39 34" src="https://github.com/zepinglee/citeproc-lua/assets/12290822/3b963862-a5f7-4295-ba39-53c3273c0591">
